main/src/library/FilePathWithError.flix

```diff
@@ -0,0 +1,481 @@
+    ///
+    /// Creates `f` if it does not exist. Overwrites it if it exists.
+    ///
+    pub def writeLines(data: {lines = List[String]}, f: String): Result[IOError, Unit]
+
+    ///
+    /// Writes `data` to the given file `f`.
+    ///
+    /// Creates file `f` if it does not exist. Overwrites it if it exists.
+    ///
+    pub def writeBytes(data: Vector[Int8], f: String): Result[IOError, Unit]
+
+    ///
+    /// Appends `str` to the given file `f`.
+    ///
+    /// Creates the file `f` if it does not exist.
+    ///
+    pub def append(data: {str = String}, f: String): Result[IOError, Unit]
+
+    ///
+    /// Appends `lines` to the given file `f`.
+    ///
+    /// Creates the file `f` if it does not exist.
+    ///
+    pub def appendLines(data: {lines = List[String]}, f: String): Result[IOError, Unit]
+
+    ///
+    /// Appends `data` to the given file `f`.
+    ///
+    /// Creates the file `f` if it does not exist.
+    ///
+    pub def appendBytes(data: Vector[Int8], f: String): Result[IOError, Unit]
+
+    ///
+    /// Truncates the given file `f`.
+    ///
+    pub def truncate(f: String): Result[IOError, Unit]
+
+    ///
+    /// Creates the directory `d`.
+    ///
+    pub def mkDir(d: String): Result[IOError, Unit]
+
+    ///
+    /// Creates the directory `d` and all its parent directories.
+    ///
+    pub def mkDirs(d: String): Result[IOError, Unit]
+
+    ///
+    /// Returns a list with the names of all files and directories in the given directory `d`.
+    ///
+    pub def list(f: String): Result[IOError, List[String]]
+
+}
+
+mod FilePath {
+
+    import java.io.File
+    import java.io.IOException
+    import java.lang.Class
+    import java.lang.{String => JString}
+    import java.lang.UnsupportedOperationException
+    import java.nio.charset.StandardCharsets
+    import java.nio.file.attribute.BasicFileAttributes
+    import java.nio.file.attribute.FileAttribute
+    import java.nio.file.FileAlreadyExistsException
+    import java.nio.file.Files
+    import java.nio.file.InvalidPathException
+    import java.nio.file.LinkOption
+    import java.nio.file.NotDirectoryException
+    import java.nio.file.OpenOption
+    import java.nio.file.Path
+    import java.nio.file.Paths
+    import java.nio.file.StandardOpenOption
+    import java.util.{List => JList}
+    import java.util.stream.Collectors
+
+    ///
+    /// Handles the `FilePath` effect of the given function `f`.
+    ///
+    /// In other words, re-interprets the `FilePath` effect using the `IO` effect.
+    ///
+    pub def handle(f: a -> b \ ef): a -> b \ (ef - FilePath) + IO = x ->
+        try {
+            Ok(f(x))
+        } with FilePath {
+            def exists(f, k) = {
+                let res = try {
+                    Ok(Files.exists(Paths.get(f), (...{}: Vector[LinkOption])))
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                };
+                k(res)
```

Comment:
I am thinking it would be cleaner with out the res local variable. Then you can immediately see `k(Ok(...))` or `k(Err(...))``

Reply:
But we can do this refactor as part of the split.